### PR TITLE
WIP - Process tracker

### PR DESCRIPTION
**Pull request by Omri-Levy**
- feat(ui): wip - started work on process tracker
- feat(*): checkpoint
- feat(ui): added an accordion card component for the process tracker
- feat(*): wip - initial implementation of process tracker in the backoffice
- feat(*): mapped collection flow steps to the process tracker
- feat(*): wip - checkpoint
- feat(*): updated instances of enums and new config

### Description
Elaborate on the subject, motivation, and context.

### Related issues
 * Provide a link to each related issue.

### Breaking changes
 * Describe the breaking changes that this pull request introduces.

### How these changes were tested
 * Describe the tests that you ran to verify your changes, including devices, operating systems, browsers and versions.

### Examples and references
 * Links, screenshots, and other resources related to this change.

### Checklist
- [] I have read the [contribution guidelines](CONTRIBUTING.md) of this project
- [] I have read the [style guidelines](STYLE_GUIDE.md) of this project
- [] I have performed a self-review of my own code
- [] I have commented my code, particularly in hard-to-understand areas
- [] I have made corresponding changes to the documentation
- [] My changes generate no new warnings and errors
- [] New and existing tests pass locally with my changes

### CODE REVIEW

File: apps/backoffice-v2/src/domains/workflow-definitions/fetchers.ts [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    .object({
+      // transitionSchema: z
+      //   .array(
+      //     z.object({
+      //       state: z.string(),
+      //       schema: z.record(z.any(), z.unknown()),
+      //       additionalParameters: z.record(z.any(), z.unknown()).optional(),
+      //     }),
+      //   )
+      //   .nullable(),
```
Comment (Omri-Levy) [inline apps/backoffice-v2/src/domains/workflow-definitions/fetchers.ts]: Copied from https://github.com/ballerine-io/ballerine/pull/1949 - commented until that PR is in.

File: apps/kyb-app/src/pages/CollectionFlow/CollectionFlow.tsx [hunk 2]
```diff
@@ -47,6 +48,36 @@ const elems = {
   divider: Divider,
 };
 
+// TODO: Find a way to make this work via the workflow-browser-sdk `subscribe` method.
+export const useCompleteLastStep = () => {
```
Comment (Omri-Levy) [inline apps/kyb-app/src/pages/CollectionFlow/CollectionFlow.tsx:52]: Previously the last step of the collection flow never gotten to `isCompleted: true`.

File: packages/ui/.storybook/main.ts [hunk 3]
```diff
@@ -8,7 +8,6 @@ const config: StorybookConfig = {
     '@storybook/addon-links',
     '@storybook/addon-essentials',
     '@storybook/addon-interactions',
-    '@storybook/addon-styling',
```
Comment (Omri-Levy) [inline packages/ui/.storybook/main.ts:11]: In recent versions of Storybook this plugin is no longer needed. Removed due to it causing it error related to PostCSS. There doesn't seem to be a negative effect due to removing it.

File: packages/ui/src/components/molecules/AccordionCard/types.ts [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+};
+
+export type AccordionCardContentProps = Omit<
+  ComponentProps<typeof Accordion>,
+  'type' | 'value' | 'onValueChange'
+> & {
+  cardProps?: ComponentProps<typeof CardContent>;
+};
+
+export type AccordionTitle = ComponentProps<typeof CardTitle> & {
```
Comment (Omri-Levy) [inline packages/ui/src/components/molecules/AccordionCard/types.ts:29]: For some reason using an interface over a type causes an error.

File: packages/ui/tailwind.config.cjs [hunk 5]
```diff
@@ -4,7 +4,6 @@ const { fontFamily } = require('tailwindcss/defaultTheme');
 module.exports = {
   darkMode: ['class'],
   content: ['./index.html', './src/**/*.{js,ts,jsx,tsx}'],
-  important: true,
```
Comment (Omri-Levy) [inline packages/ui/tailwind.config.cjs:7]: `important: true` caused the accordion animation to break. There does not seem to be a negative effect so far.

File: services/workflows-service/src/business/business.module.ts [hunk 6]
```diff
@@ -54,6 +56,8 @@ import { WorkflowTokenRepository } from '@/auth/workflow-token/workflow-token.re
     SalesforceIntegrationRepository,
     WorkflowTokenService,
     WorkflowTokenRepository,
+    UiDefinitionRepository,
+    UiDefinitionService,
```
Comment (Omri-Levy) [inline services/workflows-service/src/business/business.module.ts:60]: Could use help with using the `UIModule` instead. Kept getting different errors.

File: services/workflows-service/src/workflow/schemas/zod-schemas.ts [hunk 7]
```diff
@@ -12,18 +12,19 @@ export const SubscriptionSchema = z.discriminatedUnion('type', [
 
 export const ConfigSchema = z
   .object({
+    isCaseOverviewEnabled: z.boolean().optional(),
+    isLegacyReject: z.boolean().optional(),
+    isLockedDocumentCategoryAndType: z.boolean().optional(),
+    isDemo: z.boolean().optional(),
+    isExample: z.boolean().optional(), // OSS only
```
Comment (Omri-Levy) [inline services/workflows-service/src/workflow/schemas/zod-schemas.ts:19]: `isCaseOverviewEnabled` is the new flag, the rest are just re-ordered to be grouped by name.

File: apps/backoffice-v2/src/common/components/molecules/ProcessTracker/interfaces.ts [hunk 8]
```diff
@@ -0,0 +1,15 @@
+import { TWorkflowById } from '@/domains/workflows/fetchers';
+import { TWorkflowDefinitionById } from '@/domains/workflow-definitions/fetchers';
+
+export interface IProcessTrackerProps {
+  tags: TWorkflowById['tags'];
+  plugins: Array<
+    | NonNullable<NonNullable<TWorkflowDefinitionById['extensions']>['apiPlugins']>[number]
```
Comment (tomer-shvadron) [inline apps/backoffice-v2/src/common/components/molecules/ProcessTracker/interfaces.ts:7]: This type is very complex and hard to understand. Will be better if inner types will be extracted and named 😄

File: apps/backoffice-v2/src/common/components/molecules/ProcessTracker/ProcessTracker.tsx [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+    childWorkflows,
+  });
+
+  return (
+    <div className={`max-w-xs`}>
+      <AccordionCard value={uncollapsedItemValue} onValueChange={onValueChange}>
+        <AccordionCard.Title>Processes</AccordionCard.Title>
+        <AccordionCard.Content>
+          <AccordionCard.Item
+            title={`Collection flow`}
```
Comment (tomer-shvadron) [inline apps/backoffice-v2/src/common/components/molecules/ProcessTracker/ProcessTracker.tsx:73]: I think we should de-couple the process tracker from it's current use.
For instance - extract items to a prop with an interface containing title, value and sub-items

File: apps/backoffice-v2/src/common/components/molecules/ProcessTracker/hooks/useProcessTrackerLogic/interfaces.ts [hunk 10]
```diff
@@ -0,0 +1,15 @@
+import { TWorkflowById } from '@/domains/workflows/fetchers';
+import { TWorkflowDefinitionById } from '@/domains/workflow-definitions/fetchers';
+
+export interface IUseProcessTrackerLogicParams {
+  tags: TWorkflowById['tags'];
+  plugins: Array<
+    | NonNullable<NonNullable<TWorkflowDefinitionById['extensions']>['apiPlugins']>[number]
```
Comment (tomer-shvadron) [inline apps/backoffice-v2/src/common/components/molecules/ProcessTracker/hooks/useProcessTrackerLogic/interfaces.ts:7]: Can we simplify this type-hell?

File: apps/kyb-app/src/components/organisms/DynamicUI/StateManager/StateManager.tsx [hunk 11]
```diff
@@ -16,6 +22,11 @@ export const StateManager = ({
   workflowId,
   initialContext,
 }: StateManagerProps) => {
+  const { stateApi } = useStateManagerContext();
```
Comment (tomer-shvadron) [inline apps/kyb-app/src/components/organisms/DynamicUI/StateManager/StateManager.tsx]: I don't understand the need in lines 25-29 as their products are not used anywhere inside or outside StateManager (they are not exported)

**Review by tomer-shvadron — APPROVED** [review]

File: apps/backoffice-v2/src/common/components/molecules/ProcessTracker/ProcessTracker.tsx (hunk 9 above)
Comment (Omri-Levy) [reply apps/backoffice-v2/src/common/components/molecules/ProcessTracker/ProcessTracker.tsx:73]: That's the design of `AccordionCard`. If you think of `ProcessTracker` as the implementation of an `AccordionCard` it sounds like your suggestion is already in place but just encapsulated.

File: apps/backoffice-v2/src/common/components/molecules/ProcessTracker/hooks/useProcessTrackerLogic/interfaces.ts (hunk 10 above)
Comment (Omri-Levy) [reply apps/backoffice-v2/src/common/components/molecules/ProcessTracker/hooks/useProcessTrackerLogic/interfaces.ts:7]: Not as part of this PR.

File: apps/backoffice-v2/src/common/components/molecules/ProcessTracker/ProcessTracker.tsx (hunk 9 above)
Comment (Omri-Levy) [reply apps/backoffice-v2/src/common/components/molecules/ProcessTracker/ProcessTracker.tsx:73]: Revisit once there's a second instance of the process tracker.

File: apps/backoffice-v2/src/common/components/molecules/ProcessTracker/interfaces.ts (hunk 8 above)
Comment (Omri-Levy) [reply apps/backoffice-v2/src/common/components/molecules/ProcessTracker/interfaces.ts:7]: Will be revisited soon.

**Pull request merged**